client/my-sites/upgrades/checkout-thank-you/jetpack-thank-you-card.jsx

```diff
@@ -387,31 +507,52 @@ class JetpackThankYouCard extends Component {
 		} );
 	}
 
-	renderAction() {
+	getProgress() {
+		if ( this.isErrored() ) {
+			return 0;
+		}
+
 		const features = this.getFeaturesWithStatus() || [ '' ];
 		const completed = this.shouldRenderPlaceholders()
 			? 0
 			: reduce( features, ( total, feature ) => {
-				if ( 'object' !== typeof feature || ! feature.hasOwnProperty( 'status' ) || ! feature.status ) {
+				if ( 'object' !== typeof feature || ! feature.hasOwnProperty( 'status' ) || 'done' !== feature.status ) {
 					return total;
 				}
 
 				return total += 1;
 			}, 0 );
 
-		if ( completed === features.length ) {
+		return Math.ceil( completed / features.length * 100 );
+	}
+
+	renderAction( progress = 0 ) {
+		// We return an empty span for the error case becaue the default button will be displayed
+		// further down the tree if the action is falsey.
+		if ( this.isErrored() ) {
+			return <span />;
+		}
+
+		if ( 100 === progress ) {
 			return null;
 		}
 
 		return (
-			<ProgressBar value={ completed } total={ features.length } isPulsing />
+			<ProgressBar value={ progress } isPulsing />
 		);
 	}
 
+	renderDescription( progress = 0 ) {
+		const { translate } = this.props;
+		if ( 100 === progress ) {
+			return translate( "You are powered up, it's time to see your site." );
+		}
+
+		return translate( "Now that we've taken care of the plan, it's time to power up your site." );
```

Comment:
Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Now that we've taken care of the plan, it's time to see your new site.' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).